src/client.js

```diff
@@ -67,13 +67,13 @@ function Client(manager, name, config) {
 
 	var client = this;
 
-	if (!client.config.token) {
-		client.updateToken(function() {
-			client.manager.updateUser(client.name, {token: client.config.token});
-		});
-	}
-
 	if (config) {
+		if (config.token) {
```

Comment:
I guess the purpose of this PR is to ensure that the `config` (or `client.config` depending on where it is used) is not `null` or `undefined` so that `config.something` / `client.config.something` does not crash the app.

However, why did you move from `!client.config.token` to, effectively, `client.config.token`?